### PR TITLE
Update the service account used by private Prow jobs

### PR DESCRIPTION
**Pull request by chizhg**

### CODE REVIEW

**Review by chaodaiG — APPROVED** [review]
/hold
bind SA before unhold

**Comment by chizhg** [comment]
> /hold bind SA before unhold

Do you mean the config change in prow/cluster/build/prowjob_service_accounts.yaml? Shouldn't it be updated automatically by the prow-deployer postsubmit job?

File: prow/cluster/build/prowjob_service_accounts.yaml [hunk 1]
```diff
@@ -11,6 +11,16 @@ metadata:
 ---
 apiVersion: v1
 kind: ServiceAccount
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: istio-prow-test-job-private@istio-testing.iam.gserviceaccount.com
+  labels:
+    app.kubernetes.io/part-of: prow
+  namespace: test-pods
+  name: prowjob-private-sa
```
Comment (howardjohn) [inline prow/cluster/build/prowjob_service_accounts.yaml:24]: comments of what permissions are allowed would probably help. Or at least the intended usage

**Review by howardjohn — APPROVED** [review]

**Comment by chaodaiG** [comment]
> > /hold bind SA before unhold
> 
> Do you mean the config change in prow/cluster/build/prowjob_service_accounts.yaml? Shouldn't it be updated automatically by the prow-deployer postsubmit job?

I mean binding workload identity with gcloud

**Comment by chizhg** [comment]
> > > /hold bind SA before unhold
> > 
> > 
> > Do you mean the config change in prow/cluster/build/prowjob_service_accounts.yaml? Shouldn't it be updated automatically by the prow-deployer postsubmit job?
> 
> I mean binding workload identity with gcloud

I forgot that.. Done.

File: prow/cluster/build/prowjob_service_accounts.yaml (hunk 1 above)
Comment (chizhg) [reply prow/cluster/build/prowjob_service_accounts.yaml:24]: I've added the comments. Please take a second look.

**Pull request merged**